api/src/main/java/org/openmrs/api/db/SerializedObject.java

```diff
@@ -9,22 +9,45 @@
  */
 package org.openmrs.api.db;
 
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.annotations.Parameter;
 import org.openmrs.BaseChangeableOpenmrsMetadata;
 import org.openmrs.serialization.OpenmrsSerializer;
 
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
 /**
  * Object representation of a Serialized Object as stored in the database.
  */
+@Entity
+@Table(name = "serialized_object")
 public class SerializedObject extends BaseChangeableOpenmrsMetadata {
-	
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "serialized_object_id_seq")
+	@GenericGenerator(
+		name = "serialized_object_id_seq",
+		strategy = "native",
+		parameters = @Parameter(name = "sequence", value = "serialized_object_serialized_object_id_seq")
+	)
+	@Column(name = "serialized_object_id")
 	private Integer id;
-	
+
+	@Column(name = "type", nullable = false)
```

Comment:
Shouldn't these strings have a length attribute?

Reply:
added string lengths

Reply:
@dkayiwa  any update on this?